### PR TITLE
@W-14781712 fix: added null checks

### PR DESCRIPTION
**Pull request by abbasmanthirik**
Added null checks for token

### CODE REVIEW

**Review by sahil-here — CHANGES_REQUESTED** [review]
We have a few queries:
**Code Change related:**
1) Why a null check has been added here for W-14781712 where EntityFormulaContextBase.createMockFieldInfo is throwing the "IllegalArgumentException: entity Info is null"? How is this check related to that exception?
**Code Quality related:**
2) Why there is a need for the "else" block?
3) Why the same check has not been added to the another constructor of the same class?
4) Why has the datatype been updated from primitive to object for location and type? If yes, then does the getter/setter need to be updated or not, has that been considered?

**Comment by abbasmanthirik** [comment]
> We have a few queries: **Code Change related:**
> 
> 1. Why a null check has been added here for W-14781712 where EntityFormulaContextBase.createMockFieldInfo is throwing the "IllegalArgumentException: entity Info is null"? How is this check related to that exception?
>    **Code Quality related:**
> 2. Why there is a need for the "else" block?
> 3. Why the same check has not been added to the another constructor of the same class?
> 4. Why has the datatype been updated from primitive to object for location and type? If yes, then does the getter/setter need to be updated or not, has that been considered?

1. The actual root cause for the gack is below from the stack trace link

`
Cause0: | java.lang.NullPointerException
-- | --
Cause0-StackTrace: | at com.force.formula.impl.WrongArgumentTypeException.<init>(WrongArgumentTypeException.java:23)at com.force.formula.impl.AnnotationVisitor.verifyInputs(AnnotationVisitor.java:114)at com.force.formula.impl.AnnotationVisitor.visit(AnnotationVisitor.java:82)at com.force.formula.impl.BaseFormulaInfoImpl.visitInorderIterative(BaseFormulaInfoImpl.java:746)at com.force.formula.impl.BaseFormulaInfoImpl.visit(BaseFormulaInfoImpl.java:732)at com.force.formula.impl.BaseFormulaInfoImpl.annotateAst(BaseFormulaInfoImpl.java:484)at com.force.formula.impl.BaseFormulaInfoImpl.<init>(BaseFormulaInfoImpl.java:117)at common.formula.FormulaInfoImpl.<init>(FormulaInfoImpl.java:61)at common.formula.FormulaInfoImpl.<init>(FormulaInfoImpl.java:56)at common.formula.FormulaCoreFactory.create(FormulaCoreFactory.java:84)at com.force.formula.impl.FormulaInfoFactory.create(FormulaInfoFactory.java:89)at common.formula.template.FormulaTemplate.<init>(FormulaTemplate.java:149)
`
2. If those properties(location, type, text) are left uniinitialized without else block, there will be NPE furhter downstream
3. Done
4. Switched to primitive

File: impl/src/main/java/com/force/formula/impl/WrongArgumentTypeException.java [hunk 1]
```diff
@@ -64,8 +70,13 @@ public WrongArgumentTypeException(String function, Type[] expectedInputTypes, Fo
 
         Token token = actual.getToken();
         location = token.getColumn();
```
Comment (sahil-here) [inline impl/src/main/java/com/force/formula/impl/WrongArgumentTypeException.java:72]: Why is the "location" outside the null check here?

Comment (abbasmanthirik) [reply impl/src/main/java/com/force/formula/impl/WrongArgumentTypeException.java]: Was a miss. Fixed it. Thanks

Comment (abbasmanthirik) [reply impl/src/main/java/com/force/formula/impl/WrongArgumentTypeException.java]: > Why is the "location" outside the null check here?
Done

**Review by sahil-here — APPROVED** [review]

**Pull request closed without merging**